tests/integration/test_charm.py

```diff
@@ -2,122 +2,140 @@
+
+
+def fix_queryselector(elems):
+    """Workaround for web components breaking querySelector.
+    Because someone thought it was a good idea to just yeet the moral equivalent
+    of iframes everywhere over a single page 🤦
+    Shadow DOM was a terrible idea and everyone involved should feel professionally
+    ashamed of themselves. Every problem it tried to solved could and should have
+    been solved in better ways that don't break the DOM.
+    """
+
+    selectors = '").shadowRoot.querySelector("'.join(elems)
+    return 'return document.querySelector("' + selectors + '")'
+
+
+@pytest.mark.asyncio
 @pytest.mark.abort_on_fail
-async def test_build_and_deploy(ops_test):
+async def test_build_and_deploy(ops_test: OpsTest):
     my_charm = await ops_test.build_charm(".")
     image_path = METADATA["resources"]["oci-image"]["upstream-source"]
 
-    await ops_test.model.deploy(my_charm, resources={"oci-image": image_path})
+    await ops_test.model.deploy(
+        my_charm, resources={"oci-image": image_path}, trust=True
+    )
 
-    charm_name = METADATA["name"]
     await ops_test.model.wait_for_idle(
-        [charm_name],
+        [CHARM_NAME],
         raise_on_blocked=True,
         raise_on_error=True,
         timeout=300,
     )
-    assert ops_test.model.applications[charm_name].units[0].workload_status == "waiting"
+    assert ops_test.model.applications[CHARM_NAME].units[0].workload_status == "waiting"
     assert (
-        ops_test.model.applications[charm_name].units[0].workload_status_message
+        ops_test.model.applications[CHARM_NAME].units[0].workload_status_message
         == "Waiting for kubeflow-profiles relation data"
     )
 
 
+@pytest.mark.asyncio
 @pytest.mark.abort_on_fail
-async def test_add_profile_relation(ops_test):
-    charm_name = METADATA["name"]
-    await ops_test.model.deploy("kubeflow-profiles", channel="latest/edge")
-    await ops_test.model.add_relation("kubeflow-profiles", charm_name)
+async def test_add_profile_relation(ops_test: OpsTest):
+    await ops_test.model.deploy(PROFILES_CHARM_NAME, channel="latest/edge", trust=True)
+    await ops_test.model.add_relation(PROFILES_CHARM_NAME, CHARM_NAME)
     await ops_test.model.wait_for_idle(
-        ["kubeflow-profiles", charm_name],
+        [PROFILES_CHARM_NAME, CHARM_NAME],
         status="active",
         raise_on_blocked=True,
         raise_on_error=True,
         timeout=300,
     )
 
 
-async def test_status(ops_test):
-    charm_name = METADATA["name"]
-    assert ops_test.model.applications[charm_name].units[0].workload_status == "active"
+@pytest.mark.asyncio
+async def test_status(ops_test: OpsTest):
+    assert ops_test.model.applications[CHARM_NAME].units[0].workload_status == "active"
 
 
-def fix_queryselector(elems):
-    """Workaround for web components breaking querySelector.
+@pytest.mark.asyncio
+async def test_configmap_exist():
+    configmap = Client().get(ConfigMap, CONFIGMAP_NAME, namespace="kubeflow")
```

Comment:
We could use a pytest fixture for the client
```python
@pytest.fixture
async def lightkube_client():
    lightkube_client = Client(field_manager="test")
    yield lightkube_client
```

Reply:
Refactored .. Thanks :)